service/resource/cloudformation/resource.go

```diff
@@ -1,72 +1,68 @@
 package cloudformation
 
 import (
-	"context"
-
-	"github.com/giantswarm/aws-operator/service/cloudconfig"
-	"github.com/giantswarm/certificatetpr"
+	awsCF "github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/aws-sdk-go/service/cloudformation/cloudformationiface"
+	awsutil "github.com/giantswarm/aws-operator/client/aws"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/micrologger"
 	"github.com/giantswarm/operatorkit/framework"
-	"k8s.io/client-go/kubernetes"
 )
 
 const (
 	// Name is the identifier of the resource.
 	Name = "cloudformation"
 )
 
-// Config represents the configuration used to create a new config map resource.
+type AWSConfig struct {
+	AccessKeyID     string
+	AccessKeySecret string
+	SessionToken    string
+	Region          string
+	accountID       string
+}
+
+// Config represents the configuration used to create a new cloudformation resource.
 type Config struct {
 	// Dependencies.
-	CertWatcher certificatetpr.Searcher
-	CloudConfig *cloudconfig.CloudConfig
-	K8sClient   kubernetes.Interface
-	Logger      micrologger.Logger
+	Clients awsutil.Clients
+	Logger  micrologger.Logger
 }
 
-// DefaultConfig provides a default configuration to create a new config map
+// DefaultConfig provides a default configuration to create a new cloudformation
 // resource by best effort.
 func DefaultConfig() Config {
 	return Config{
 		// Dependencies.
-		CertWatcher: nil,
-		CloudConfig: nil,
-		K8sClient:   nil,
-		Logger:      nil,
+		Clients: awsutil.Clients{},
+		Logger:  nil,
 	}
 }
 
-// Resource implements the config map resource.
+// Resource implements the cloudformation resource.
 type Resource struct {
 	// Dependencies.
-	certWatcher certificatetpr.Searcher
-	cloudConfig *cloudconfig.CloudConfig
-	k8sClient   kubernetes.Interface
-	logger      micrologger.Logger
+	awsClient cloudformationiface.CloudFormationAPI
+	logger    micrologger.Logger
 }
 
-// New creates a new configured config map resource.
+// StackState is the state representation pn which the resource methods work
+type StackState struct {
+	Name         string
+	TemplateBody string
+	Parameters   []*awsCF.Parameter
+}
```

Comment:
FYI so far we put things like these into a `spec.go`.

Reply:
Thx, moved there, I've also removed an unused field.